util/hook/hook.go

```diff
@@ -21,38 +22,44 @@ func IsHook(obj *unstructured.Unstructured) bool {
 }
 
 func Skip(obj *unstructured.Unstructured) bool {
-	for _, hookType := range types(obj) {
-		if v1alpha1.HookType(hookType) == v1alpha1.HookTypeSkip {
-			return len(types(obj)) == 1
+	for _, hookType := range Types(obj) {
+		if hookType == v1alpha1.HookTypeSkip {
+			return len(Types(obj)) == 1
 		}
 	}
 	return false
 }
 
 func Types(obj *unstructured.Unstructured) []v1alpha1.HookType {
-	var hookTypes []v1alpha1.HookType
-	for _, text := range types(obj) {
-		hookType := v1alpha1.HookType(text)
-		switch hookType {
-		case v1alpha1.HookTypePreSync, v1alpha1.HookTypeSync, v1alpha1.HookTypePostSync, v1alpha1.HookTypeSyncFail:
-			hookTypes = append(hookTypes, hookType)
+	typeToBool := make(map[v1alpha1.HookType]bool)
```

Comment:
I think this is wrong, currently, we treat it as a super-set of hooks, but actually, we should ignore Helm if any Argo values are set.